examples/seismic/tutorials/06_skimage_tv.ipynb

```diff
@@ -0,0 +1,416 @@
+    "def get_data(param):\n",
+    "    \"\"\" Returns source and receiver data for a single shot labeled 'shot_id'.\n",
+    "    \"\"\"\n",
+    "    true_model = get_true_model()\n",
+    "    dt = true_model.critical_dt  # Time step from model grid spacing\n",
+    "\n",
+    "    # Set up source data and geometry.\n",
+    "    nt = int(1 + (param['tn']-param['t0']) / dt)  # Discrete time axis length\n",
+    "\n",
+    "    src = RickerSource(name='src', grid=true_model.grid, f0=param['f0'],\n",
+    "                       time=np.linspace(param['t0'], param['tn'], nt))\n",
+    "    src.coordinates.data[0, :] = [30, param['shot_id']*1000./(param['nshots']-1)]\n",
+    "\n",
+    "    # Set up receiver data and geometry.\n",
+    "    nreceivers = 101  # Number of receiver locations per shot\n",
+    "    rec = Receiver(name='rec', grid=true_model.grid, npoint=nreceivers, ntime=nt)\n",
+    "    rec.coordinates.data[:, 1] = np.linspace(0, true_model.domain_size[0], num=nreceivers)\n",
+    "    rec.coordinates.data[:, 0] = 980. # 20m from the right end\n",
+    "\n",
+    "    # Set up solver - using model_in so that we have the same dt,\n",
+    "    # otherwise we should use pandas to resample the time series data. \n",
+    "    solver = AcousticWaveSolver(true_model, src, rec, space_order=4)\n",
+    "\n",
+    "    # Generate synthetic receiver data from true model\n",
+    "    true_d, _, _ = solver.forward(src=src, m=true_model.m)\n",
+    "\n",
+    "    return src, true_d, nt, solver\n",
+    "\n",
+    "# Define a type to store the functional and gradient.\n",
+    "class fg_pair:\n",
+    "    def __init__(self, f, g):\n",
+    "        self.f = f\n",
+    "        self.g = g\n",
+    "    \n",
+    "    def __add__(self, other):\n",
+    "        f = self.f + other.f\n",
+    "        g = self.g + other.g\n",
+    "        \n",
+    "        return fg_pair(f, g)\n",
+    "    \n",
+    "    def __radd__(self, other):\n",
+    "        if other == 0:\n",
+    "            return self\n",
+    "        else:\n",
+    "            return self.__add__(other)\n",
+    "\n",
+    "from devito import Function, clear_cache\n",
+    "\n",
+    "# Create FWI gradient kernel for a single shot\n",
+    "def fwi_gradient_i(param):\n",
+    "    # Need to clear the workers cache.\n",
+    "    clear_cache()\n",
+    "\n",
+    "    # Load the current model and the shot data for this worker.\n",
+    "    # Note, unlike the serial example the model is not passed in\n",
+    "    # as an argument. Broadcasting large datasets is considered\n",
+    "    # a programming anti-pattern and at the time of writing it\n",
+    "    # it only worked relaiably with Dask master. Therefore, the\n",
+    "    # the model is communicated via a file.\n",
+    "    model0 = get_current_model(param)\n",
+    "    src, rec, nt, solver = get_data(param)\n",
+    "    \n",
+    "    # Create symbols to hold the gradient and the mismatch between\n",
```

Comment:
I think you should stick to either calling it mismatch or residual. Again, you are calling it different things in different places which might be confusing to someone who doesn't use the same terminology.

Reply:
It's called residual or data misfit, not mimatch.

Reply:
fixed.